### DISCUSSION TITLE
int8 Arrays displayed as strings in Supabase Dashboard

### DISCUSSION THREAD

**Discussion by yayza** [supabase/supabase#16942] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

`int8[]` column data is always displayed as an array of strings in the Supabase Table Editor and SQL Editor

Additionally, table definitions don't specify array types and are just displayed as `array null`:

```
create table
  public.test_arrays (
    id bigint generated by default as identity not null,
    array_int2 array null,
    array_int4 array null,
    array_int8 array null,
    array_float4 array null,
    array_float8 array null,
    array_numeric array null,
    array_bool array null,
    constraint test_arrays_pkey primary key (id)
  ) tablespace pg_default;
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Using the Supabase SQL editor:
```
create table if not exists public.test_arrays (
  id bigint generated by default as identity not null, 
  array_int2 int2[] null, 
  array_int4 int4[] null, 
  array_int8 int8[] null, 
  array_float4 float4[] null, 
  array_float8 float8[] null, 
  array_numeric numeric[] null, 
  array_bool bool[] null, 
  constraint test_arrays_pkey primary key (id)
) tablespace pg_default;


INSERT into public.test_arrays(
  array_int2, array_int4, array_int8, 
  array_float4, array_float8, array_numeric, 
  array_bool
) 
VALUES 
  (
    ARRAY[1, 2], ARRAY[1, 2], ARRAY[1, 2], 
    ARRAY[1, 2], ARRAY[1, 2], ARRAY[1, 
    2], ARRAY[true, false]
  )

```

## Expected behavior

For array values to be displayed as their corresponding array type

## Screenshots

In table editor:
![image](https://i.imgur.com/l9tg221.png)

In sql editor:
![image](https://i.imgur.com/8PvsJRD.png)

**Comment by GaryAustin1** [comment]
The table definitions are problematic if the intent is for users to use that sql code.  It is not valid sql for all array type columns.

I suspect the int8 has something to do with javascript not being able to handle int8 data.  Probably some workaround was attempted,  but just a guess.

**Comment by Arjun3492** [comment]
Tried finding out if there was an issue with the code but couldn't find any from my side maybe according to @GaryAustin1 since int8 is a type of BIGINT it might be converted into strings to avoid potential loss of precision or other compatibility issues

**Comment by sh3rlock97** [comment]
when using int8 arrays you should convert them to a bytea and use the bytea column type.

Supabase supports echanging bytea as hex //x
